### ISSUE TITLE
Deprecate integer version code

### ISSUE THREAD

**Issue by dvz** [mybb/mybb#4327]
Remove, or deprecate with intent to remove, the integer version code `1900` in favor of the `1.9.0` format.

A list of known usages follows.

`mybb/mybb`:
- codebase:
  - `MyBB::$version_code`
  - `$langinfo['version']` in language manifest files
  - attributes of `<theme>`, `<stylesheet>`, `<template>` in theme XML files
  - attributes of `<adminviews>` in Admin CP's View Manager
  - plugin compatiblity string
  - uploaded upgrade scripts check
  - latest version check
  - Extend browsing API
  - checksum files API
  - installation server specifications collection
  - arbitrary browser cache-busting parameters
- other:
  - `mybb/mybb` repository tags

MyBB.com infrastructure:
- Extend browsing API
- checksum files API
- Installation Statistics

**Comment by dragonexpert** [comment]
As long as this is in 1.9 fine.  In the 1.8 series, I specifically use version code because of format_avatar requiring a specific version of the 1.8 series or higher.  I wouldn't be surprised if I am not the only one who uses $Mybb->version_code to check availability of a feature / function.

**Comment by dvz** [comment]
Yes, we always want to avoid breaking changes, and only introduce them with new series.

This should be done with either 1.9.0 or 1.10.0.

**Comment by Sama34** [comment]
@dragonexpert using `182*` in the `compatibility` line should be possible.

I think we could deprecate this in 1.8 and force it in 1.9

**Comment by cronlabspl** [comment]
You can always change every occurance of shorted form ( `1800` ) to `1.8.0.0`
